cdap-runtime-spi/src/main/java/io/cdap/cdap/runtime/spi/launcher/Launcher.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.runtime.spi.launcher;
+
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * Job launcher to prepare and launch the job.
+ */
+public interface Launcher {
+
+  /**
+   * Initializes the launcher.
```

Comment:
would be good to include more information about how this is used. I assume it is called once before any other method is called?